lib/src/signed_video_h26x_auth.c

```diff
@@ -115,6 +115,112 @@ decode_sei_data(signed_video_t *self, const uint8_t *payload, size_t payload_siz
   return status;
 }
 
+/**
+ * Computes the GOP hash from |nalu_hash_list| and compares it with the GOP hash received from the
+ * signature.
+ *
+ * @param self A pointer to the signed_video_t structure.
+ * @return true if the computed GOP hash matches the received GOP hash and the status is SV_OK;
+ * false otherwise.
+ */
+static bool
+verify_gop_hash(signed_video_t *self)
+{
+  gop_info_t *gop_info = self->gop_info;
+  const size_t hash_size = self->verify_data->hash_size;
+  svrc_t status = SV_UNKNOWN_FAILURE;
+
+  // Compute the GOP hash from |nalu_hash_list|.
+  status = compute_partial_gop_hash(
+      self, gop_info->nalu_hash_list, gop_info->nalu_list_idx, gop_info->computed_gop_hash);
+  // Compare the computed GOP hash with the received GOP hash.
+  bool hashes_match =
+      (memcmp(gop_info->computed_gop_hash, self->received_gop_hash, hash_size) == 0);
+
+  // Verify the GOP hash if both the hashes match and the status is SV_OK.
+  gop_info->successfully_verified_gop_hash = (hashes_match && (status == SV_OK));
+
+  return gop_info->successfully_verified_gop_hash;
+}
+
+/**
+ * Adds the hashes of NALUs used in computing the GOP hash and marks these NALUs as part of the GOP
+ * hash.
+ *
+ * This function iterates through the NALU list, identifies the NALUs that belong to the current
+ * GOP, and marks them as used in the GOP hash. It then copies the hashes of these NALUs to the
+ * `nalu_hash_list` for later verification of the GOP hash. Additionally, it handles specific cases
+ * such as moving linked hash values for I-frames and ensures that SEI-related NALUs are
```

Comment:
To hide another operation (in this case the linked hash update) inside a function that does something else (in this case prepares for computing the gop hash) is a little bit sneaky. It can easily introduce bugs. Either separate the operations or be even more explicit in the function name.